src/Functions/bitSlice.cpp

```diff
@@ -40,28 +38,22 @@ class FunctionBitSlice : public IFunction
 
     bool useDefaultImplementationForConstants() const override { return true; }
 
-    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
     {
-        const size_t number_of_arguments = arguments.size();
-
-        if (number_of_arguments < 2 || number_of_arguments > 3)
-            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
-                "Number of arguments for function {} doesn't match: passed {}, should be 2 or 3",
-                getName(), number_of_arguments);
-
-        if (!isString(arguments[0]) && !isStringOrFixedString(arguments[0]))
-            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of argument of function {}",
-                arguments[0]->getName(), getName());
-        if (arguments[0]->onlyNull())
-            return arguments[0];
-
-        if (!isNativeNumber(arguments[1]))
-            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of second argument of function {}",
-                arguments[1]->getName(), getName());
-
-        if (number_of_arguments == 3 && !isNativeNumber(arguments[2]))
-            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of second argument of function {}",
-                arguments[2]->getName(), getName());
+        FunctionArgumentDescriptors mandatory_args{
+            {"s", static_cast<FunctionArgumentDescriptor::TypeValidator>(&isStringOrFixedString), nullptr, "String"},
+            {"offset", static_cast<FunctionArgumentDescriptor::TypeValidator>(&isNativeNumber), nullptr, "(U)Int8, (U)Int16, (U)Int32, (U)Int64 or Float"},
```

Comment:
`(U)Int8, (U)Int16, (U)Int32, (U)Int64 or Float` --> `(U)Int* or Float*`

Reply:
Doesn't `(U)Int*` include 128 & 256 length integers? The function doesn't support those, hence the explicit references.

Reply:
Oh, you are right. To avoid repetition, maybe `(U)Int8/16/32/64 or Float`?

Reply:
That makes sense. Do you want me to open a PR to update this description (this PR was already merged)?

Reply:
That's fine, I'll create a PR with some misc stuff including this one